Add a mybinder.org tab to the link builder

GeorgianaElena opened the pull request:
*Try it out* link: https://106-92329878-gh.circle-artifacts.com/0/html/link.html
- new `mybinder.org` tab
- *Git Repository URL* becomes *Git Environment Repository URL* when in `mybinder.org` tab
- new form to add content repo and branch (disabled for tabs other than `mybinder.org`)
- branches are by default `master`

Closes https://github.com/jupyterhub/nbgitpuller/issues/125

yuvipanda commented on docs/_static/link_gen/link.js:
```diff
@@ -58,22 +84,52 @@ var apps = {
+    var id = div.id;
+
+    if (id.includes("mybinder")) {
+        hub.placeholder = "https://mybinder.org";
```
Maybe set it to default value, so folks don't have to manually edit it?

Do you think we can also change the label to say 'BinderHub URL'?

yuvipanda replied:
Also, the label under 'hub name' says 'hub must have nbgitpuller installed', but in the mybinder.org situation, the *environment repo* must have nbgitpuller installed.

yuvipanda commented on docs/_static/link_gen/link.js:
```diff
@@ -58,22 +84,52 @@ var apps = {
+        hub.placeholder = "https://hub.example.com";
+        hub.disabled = false;
+        env_repo.labels[0].innerHTML = "Git Repository URL";
+        content_repo.disabled = true;
```
What do you think about hiding rather than just disabling these in the other tabs? They only make sense for BinderHubs, and nothing else.

GeorgianaElena commented:
Thanks a lot for reviewing it @yuvipanda :hugs:
I've added 2 commits addressing your feedback.

P.S. I made the PR a WIP because I'm a total newbie in html/js and I wanted to review it another day to see what I can improve :grimacing: So, your review was extremely helpful. Thanks :heart:

yuvipanda commented:
This is awesome work, @GeorgianaElena! Congratulations on getting your first PR merged here - your HTML / JS in this PR were great :)

choldgraf commented:
amazing! and congratulations on your first PR @GeorgianaElena ;-0

meeseeksmachine commented:
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/how-to-reduce-mybinder-org-repository-startup-time/4956/16
